sdp.c

```diff
@@ -672,22 +719,19 @@ int janus_sdp_parse_candidate(void *ice_stream, const char *candidate, int trick
 				JANUS_LOG(LOG_VERB, "[%"SCNu64"] mDNS candidate ignored\n", handle->handle_id);
 				return 0;
 			}
-			struct addrinfo *info = NULL;
-			janus_network_address addr;
-			janus_network_address_string_buffer addr_buf;
-			if(getaddrinfo(rip, NULL, NULL, &info) != 0 ||
-					janus_network_address_from_sockaddr(info->ai_addr, &addr) != 0 ||
-					janus_network_address_to_string_buffer(&addr, &addr_buf) != 0) {
-				JANUS_LOG(LOG_WARN, "[%"SCNu64"] Couldn't resolve mDNS address (%s), dropping candidate\n",
-					handle->handle_id, rip);
-				if(info)
-					freeaddrinfo(info);
-				return res;
-			}
-			freeaddrinfo(info);
-			JANUS_LOG(LOG_VERB, "[%"SCNu64"] mDNS address (%s) resolved: %s\n",
-				handle->handle_id, rip, janus_network_address_string_from_buffer(&addr_buf));
-			g_strlcpy(rip, janus_network_address_string_from_buffer(&addr_buf), sizeof(rip));
+			/* We'll resolve this address asynchronously, in order not to keep this thread busy */
+			JANUS_LOG(LOG_VERB, "[%"SCNu64"] Resolving mDNS address (%s) asynchronously\n",
+				handle->handle_id, rip);
+			janus_sdp_mdns_candidate *mc = g_malloc(sizeof(janus_sdp_mdns_candidate));
+			janus_refcount_increase(&handle->ref);
+			mc->handle = handle;
+			mc->candidate = g_strdup(candidate);
+			mc->local = g_strdup(rip);
+			mc->cancellable = g_cancellable_new();
```

Comment:
1) No need to handle references to `cancellable` when the task has ended?
2) Are we leaving to future enhancements the capability to cancel an async task instead of e.g. waiting for it to complete before refcounting the handle to 0 ? In this case we could set the `mc` field to `NULL` for now.